SEC/readSEC.js

```diff
@@ -91,130 +85,342 @@ const axios = require("axios");
-      (el) => el.innerText,
-      (
-        await page.$x(
-          "//*[@id='" + meterlist[i].puppeteerSelector + "']/td[1]/a",
-        )
-      )[0],
-    );
-
-    const totalYieldYesterdayElement = await page.$x(
-      "//*[@id='" + meterlist[i].puppeteerSelector + "']/td[3]",
-    );
-    const totalYieldYesterday = await page.evaluate(
-      (el) => el.innerText.replace(",", ""),
-      totalYieldYesterdayElement[0],
-    );
-
-    const PVTable = {
-      meterName,
-      meterID,
-      time,
-      time_seconds,
-      PVSystem,
-      totalYieldYesterday,
-    };
-
-    PV_tableData.push(PVTable);
+}
+
+/**
+  * Returns a boolean indicating whether the selected month matches the given month
+ */
+async function isCorrectMonth(month) {
+  month = parseInt(month);
+  
+  // wait for the selected date text to appear in the table
+  const selectedDateText = await page.$eval(
+    `${DATA_TABLE} tr:nth-child(2) td:first-child`,
+    el => el.textContent.trim()
+  );
+  const selectedMonthText = selectedDateText.split("/")[1]; // extract the month
+  
+  return parseInt(selectedMonthText) === month;
+}
+
+/**
+ * Gets the daily data for a given date and adds it to the PV_tableData array
+ */
+async function getDailyData(date, meterName, meterID, PVSystem) {
+  const { END_TIME, END_TIME_SECONDS, SEC_YEAR, SEC_MONTH, SEC_DAY, SEC_DATE } = formatDateAndTime(date);
+
+  if (!await isCorrectMonth(SEC_MONTH)) {
+    await selectPreviousMonthIfNeeded(SEC_YEAR, SEC_MONTH);
   }
+  
+  let monthFlag = false; // flag to check if the month has been found
+  let dayCheck = parseInt(SEC_DAY); // day to check in the table
+  let totalDailyYield = "0";
+
+  // no point in checking multiple attempts, if the frontend state didn't load it's already too late
+  // for now just add a big timeout after clicking each of the "Details" / "Monthly" tabs
+  // potential TODO: identify loading animations and wait for those to disappear, or some other monthly indicator
+  while (!monthFlag) {
+    try {      
+      // get the total yield for the given day
+      const yieldRowSelector = `${DATA_TABLE}
+      tr:nth-child(${dayCheck + 1}) 
+      td:nth-child(2)`;
+      totalDailyYield = await page.$eval(
+        yieldRowSelector,
+      (el) => el.textContent.trim()
+      );
+  
+      // remove any commas if they exist so that parseFloat can handle values over 1,000
+      totalDailyYield = totalDailyYield.replace(/,/g, "");
+
+      // verify table date matches the date we are looking for
+      const dateRowSelector = `${DATA_TABLE}
+      tr:nth-child(${dayCheck + 1}) 
+      td:nth-child(1)`;
+      actualDate = await page.$eval(
+        dateRowSelector,
+      (el) => el.textContent.trim()
+      );
+      console.log("Actual Date: " + actualDate);
+
+      // create the PVTable object
+      const PVTable = {
+        meterName,
+        meterID,
+        END_TIME,
+        END_TIME_SECONDS,
+        PVSystem,
+        totalDailyYield,
```

Comment:
```suggestion
        time: END_TIME,
        time_seconds: END_TIME_SECONDS,
        PVSystem,
        totalYield: totalDailyYield,
```

Make sure this is formatted correctly for the backend!